sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -576,6 +576,7 @@ predicate
     : NOT? kind=BETWEEN lower=valueExpression AND upper=valueExpression
     | NOT? kind=IN '(' expression (',' expression)* ')'
     | NOT? kind=IN '(' query ')'
+    | comparisonOperator kind=(ANY | SOME) '(' query ')'
```

Comment:
Hi, @francis0407 . I have some questions beyond of the current implementation of this PR. Currently, this PR aims to add `ANY/SOME` over a query. 
1. Are we going to extend this to 'ALL' later? e.g. `SELECT 1 > ALL (SELECT 0)`  (PostgreSQL example)
2. Are we going to support `array`, too? e.g. `SELECT 1 > ANY (ARRAY[0,1])` (PostgreSQL example)

Reply:
1. Yes, we will extend this to 'ALL'. And it is easy to implement based on this PR.
2. I think it has not much relation with this PR. This PR focuses on subquery which will be rewrote as Join. Like `InSet`, 'ANY' with `array` will use code generation for computing. They are totally different. And IMO, it's ok to support.